pkg/api/api.go

```diff
@@ -255,16 +256,17 @@ func (a *API) RegisterQuerier(queryable storage.Queryable, engine *promql.Engine
 	api := v1.NewAPI(
 		engine,
 		queryable,
-		querier.DummyTargetRetriever{},
-		querier.DummyAlertmanagerRetriever{},
+		func(context.Context) v1.TargetRetriever { return &querier.DummyTargetRetriever{} },
```

Comment:
@gotjosh Once this PR is merged, you should be able to work on your metadata API refactoring.

Reply:
Thanks for the head's up, pretty excited to see this. We'll also be able to remove all the copy/paste for the alerts/rules API 😄